storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -196,19 +197,21 @@ public static Executor mkExecutor(WorkerState workerState, List<Long> executorId
             executor.stats = new BoltExecutorStats(ConfigUtils.samplingRate(executor.getStormConf()),ObjectReader.getInt(executor.getStormConf().get(Config.NUM_STAT_BUCKETS)));
         }
 
+        int minId = Integer.MAX_VALUE;
         Map<Integer, Task> idToTask = new HashMap<>();
         for (Integer taskId : taskIds) {
+            minId = Math.min(minId, taskId);
             try {
                 Task task = new Task(executor, taskId);
-                executor.sendUnanchored(
-                        task, StormCommon.SYSTEM_STREAM_ID, new Values("startup"), executor.getExecutorTransfer());
+                task.sendUnanchored( StormCommon.SYSTEM_STREAM_ID, new Values("startup"), executor.getExecutorTransfer(), null); // TODO: Roshan: does this get delivered/handled anywhere ?
```

Comment:
@HeartSaVioR,  @revans2 , others : would you know why we have this call to sendUnanchored ? I recall verifying on master (and 2306) that it doesn't get delivered to anywhere. Could remove it if not needed.